### ISSUE TITLE
Mouse Location Moves If Rotating While Placing Tile

### ISSUE THREAD

**Issue by MrMonop** [Bouncyrock/TaleSpire-Alpha-Public-Issue-Tracker#53]
**Describe the bug**
While placing a tile, I rotated the screen. Upon completion of the rotation, my mouse position completely changes, sometimes to off the screen (I assume where the mouse would be based on my scroll). If I'm placing the tile and I release, the now unknown area of selection has tiles placed on it. 

**To Reproduce**
1. Select Tile in Build Mode (I was using Castle Tiles)
2. When placing, press and hold left mouse, then press and hold middle mouse button to rotate.
3. Release middle mouse button to end rotation
4. Watch mouse location jump to new spot. (If left mouse was clicked and now released, you have a large area with tiles. I've repeated with pillars as well.)

**Screenshots/Videos/Twitch Clips**

[TS-Mouse-Location-Moves-During-Rotation.zip](https://github.com/Bouncyrock/TaleSpire-Alpha-Public-Issue-Tracker/files/2752735/TS-Mouse-Location-Moves-During-Rotation.zip)

**Additional context**
I was using Windowed Mode (have not tried to recreate in fullscreen mode)
I have replicated with many Castle pieces.
I assume that when rotating, my mouse location is not remaining fixed where I left clicked. Would not be an issue if I were not trying to place tiles.

**Comment by cbaggers** [comment]
Yeah this is an annoying one. We are using the hardware cursor so we need to add some code to reposition the cursor to where makes sense.

Thanks for all the info. This is a great breakdown of the issue

**Comment by cbaggers** [comment]
https://github.com/Bouncyrock/TaleSpire/issues/469

**Comment by cbaggers** [comment]
First part of this issue is adding control for hardware cursor. I'm testing that now

**Comment by cbaggers** [comment]
well that was easy. We can now start tackling this properly
